src/proxy/query.go

```diff
@@ -243,65 +252,76 @@ func (spanner *Spanner) ComQuery(session *driver.Session, query string, bindVari
 			}
 		}
 	case *sqlparser.Union:
 		if qr, err = spanner.handleSelect(session, query, node); err != nil {
 			log.Error("proxy.union[%s].from.session[%v].error:%+v", query, session.ID(), err)
+			status = 1
 		}
-		spanner.auditLog(session, W, xbase.UPDATE, query, qr)
+		spanner.auditLog(session, W, xbase.UPDATE, query, qr, status)
 		return returnQuery(qr, callback, err)
 	case *sqlparser.Kill:
 		if qr, err = spanner.handleKill(session, query, node); err != nil {
 			log.Error("proxy.kill[%s].from.session[%v].error:%+v", query, session.ID(), err)
+			status = 1
 		}
-		spanner.auditLog(session, R, xbase.KILL, query, qr)
+		spanner.auditLog(session, R, xbase.KILL, query, qr, status)
 		return returnQuery(qr, callback, err)
 	case *sqlparser.Explain:
 		if qr, err = spanner.handleExplain(session, query, node); err != nil {
 			log.Error("proxy.explain[%s].from.session[%v].error:%+v", query, session.ID(), err)
+			status = 1
 		}
-		spanner.auditLog(session, R, xbase.EXPLAIN, query, qr)
+		spanner.auditLog(session, R, xbase.EXPLAIN, query, qr, status)
 		return returnQuery(qr, callback, err)
 	case *sqlparser.Transaction:
 		// Support for myloader.
 		// Support Multiple-statement Transaction
 		if qr, err = spanner.handleMultiStmtTxn(session, query, node); err != nil {
 			log.Error("proxy.transaction[%s].from.session[%v].error:%+v", query, session.ID(), err)
+			status = 1
 		}
-		spanner.auditLog(session, R, xbase.TRANSACTION, query, qr)
+		spanner.auditLog(session, R, xbase.TRANSACTION, query, qr, status)
 		return returnQuery(qr, callback, err)
 	case *sqlparser.Radon:
 		if qr, err = spanner.handleRadon(session, query, node); err != nil {
 			log.Error("proxy.admin[%s].from.session[%v].error:%+v", query, session.ID(), err)
+			status = 1
 		}
-		spanner.auditLog(session, R, xbase.RADON, query, qr)
+		spanner.auditLog(session, R, xbase.RADON, query, qr, status)
 		return returnQuery(qr, callback, err)
 	case *sqlparser.Set:
 		log.Warning("proxy.query.set.query:%s", query)
 		if qr, err = spanner.handleSet(session, query, node); err != nil {
 			log.Error("proxy.set[%s].from.session[%v].error:%+v", query, session.ID(), err)
+			status = 1
 		}
-		spanner.auditLog(session, R, xbase.SET, query, qr)
+		spanner.auditLog(session, R, xbase.SET, query, qr, status)
 		return returnQuery(qr, callback, err)
 	case *sqlparser.Checksum:
 		log.Warning("proxy.query.checksum.query:%s", query)
 		if qr, err = spanner.handleChecksumTable(session, query, node); err != nil {
 			log.Error("proxy.checksum[%s].from.session[%v].error:%+v", query, session.ID(), err)
+			status = 1
 		}
-		spanner.auditLog(session, R, xbase.CHECKSUM, query, qr)
+		spanner.auditLog(session, R, xbase.CHECKSUM, query, qr, status)
 		return returnQuery(qr, callback, err)
 	default:
 		log.Error("proxy.unsupported[%s].from.session[%v]", query, session.ID())
-		spanner.auditLog(session, R, xbase.UNSUPPORT, query, qr)
+		spanner.auditLog(session, R, xbase.UNSUPPORT, query, qr, 1)
```

Comment:
Here we should avoid magic numbers, `1` should be replaced to `status`, next:
```
spanner.auditLog(session, R, xbase.UNSUPPORT, query, qr, 1)
```
split two steps:
```
status=1
spanner.auditLog(session, R, xbase.UNSUPPORT, query, qr, status)
```